### PR TITLE
fix needing a second container for the update

### PR DESCRIPTION
**Pull request by jakecoffman**
Instead of running the fetch_files in one container, then the update_files in another, we can just run one container that does both!

This has the following benefits:
- Simpler
  - I deleted a bunch of code! There might be more I missed too.
  - No need to mount the repo dir.
  - No need to read the output file and turn it into an input file for the next step.
- A little bit faster.
- We can remove [this code](https://github.com/dependabot/dependabot-core/blob/8a7113652960b117de8d05d4ec4f8fecc1931b5c/updater/lib/dependabot/file_fetcher_job.rb#L104-L113) from the updater that is deleting the repo afterwards since it doesn't need to be mounted anymore. This is preventing us from updating with a local repo in the CLI for testing.

The CLI has been running this way for a while and it seems pretty good.

### CODE REVIEW

File: dist/main/index.js
Comment (jeffwidman) [inline dist/main/index.js:75991]: What does this do? And who's the consumer? I don't see it anywhere else in this repo?

Because if `dependabot-action` is the only place using two containers (I don't know if it is), then since it's now one, does that mean this setting should be nuked instead?

**Review by jeffwidman — APPROVED** [review]
From a design perspective this makes sense to me.

From an implementation perspective this looks fine, but I don't really know typescript, so you may want to get a second review... or if you're comfortable, just ship as-is since I imagine if this breaks it'lldo so catastrophically so we'll realize it quickly during GHES testing...

File: dist/main/index.js
Comment (jakecoffman) [reply dist/main/index.js:75991]: It triggers [this behavior](https://github.com/dependabot/dependabot-core/blob/8a7113652960b117de8d05d4ec4f8fecc1931b5c/updater/lib/dependabot/file_fetcher_job.rb#L41-L49) which writes the input for `update_files` rather than having something outside of the container do it.

**Comment by jakecoffman** [comment]
@landongrindheim I added `UPDATER_ONE_CONTAINER` in July. I'll admit I don't know how customers typically update the action, but if they update the action and stay on an old version of the updater then that would be a problem. 

I suppose we should setup some kind of smoke testing here since we may be making more changes to the action soon.

**Comment by landongrindheim** [comment]
> if they update the action and stay on an old version of the updater then that would be a problem

I don't _think_ this should be an issue. This Action is responsible for managing the version of the Updater image. We've been using SHAs since [v2.1.0](https://github.com/github/dependabot-action/releases/tag/v2.1.0).

> I suppose we should setup some kind of smoke testing here

I'm 👍 to that idea, but I think manual testing is sufficient for this change.

**Comment by landongrindheim** [comment]
### ✅ Testing results

**TL;DR** Everything looks good 🎉 

---

🔎 This change was tested against a private repository set up for eight ecosystems:

image

* Each of the following runs was successful, with the exception of those marked with a ⚠️
* Two of the three ⚠️s were for pull request exceeded errors
  * This is expected behavior
* The Git submodules ⚠️ was for the submodule not being found
  * This is expected/desired behavior

**Pull request merged**
